web/modules/custom/dbcdk_community/src/Plugin/Block/ProfilesBlock.php

```diff
@@ -208,6 +250,48 @@ public function build() {
   }
 
   /**
+   * Get all quarantined profiles.
+   *
+   * @param array $profiles_filter
+   *   An array containing filter arguments.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   *
+   * @return \DBCDK\CommunityServices\Model\Profile[] $profiles
+   *   An array of Profile objects with an active quarantine.
+   */
+  protected function getQuarantinedProfiles(array $profiles_filter) {
+    // Fetch all quarantines that are active from the moment of the request.
+    $quarantined_filter = [
+      'where' => [
+        'end' => [
+          'gt' => date('Y-m-d\TH:i:s.000\Z'),
+        ],
+      ],
+    ];
+    $quarantines = (array) $this->quarantineApi->quarantineFind(json_encode($quarantined_filter));
+
+    // Reduce all the quarantines to an array of quarantined profile ids.
+    // We do this to make sure a profile id only appears once since we use these
+    // values as "Where" arguments for the next request to the service, and a
+    // profile can have multiple quarantines at the same time.
+    $quarantine_ids = array_reduce($quarantines, function($result, $quarantine) {
+      $result[$quarantine->getQuarantinedId()] = ['id' => $quarantine->getQuarantinedId()];
+      return $result;
+    });
+
+    // Reset array keys so json_encode() will handle the ids as an indexed array
+    // and not an associative array.
+    $quarantine_ids = array_values($quarantine_ids);
```

Comment:
We set the array keys to remove any duplications of user id's compared to just pushing to `$result[] = ` - it could be discussed if this is necessary and if a user would have two active quarantines at the same time, but it's defensive coding to counter that scenario (I could be convinced to NOT do this if we can agree that multiple "where id = ..." with the same user isn't a real issue here